form-regex.rst

```diff
@@ -0,0 +1,91 @@
+Using regular expressions
+============================
+
+.. _regex:
+
+Regular Expressions
+---------------------
+
+A regular expression is a special text string for describing a search pattern. Regular expressions find an important use in specifying constraints in your forms. You might want to set constraints on the length, character set allowed and various other fields in the data input by the user. All this can be achieved by using regular expression constraints in your form.
+
+Some basic operations on regular expressions are:
+
```

Comment:
regex strings should be in double-backticked `literal` blocks, rather than wrapped in asterisks.
(unless there's a compelling reason to add a `:regex:` role...)

Using the asterisks is going to get confusing -- makes the source hard to read because the asterisks also have a meaning inside regex.